Standards/scs-XXXX-v1-time-sync-decisions.md

```diff
@@ -0,0 +1,122 @@
+
+It is unclear if PTP would offer any better precision than NTP in tunneled networks with virtual interfaces, and it does not seem to get any use in IaaS at all.
+
+### Paravirtualization
+
+Unlike PTP as a protocol, it's driver interface in Linux seems to be a popular choice for time synchronization in the form of paravirtualization.
+The _ptp\_kvm_ kernel module will communicate directly with the KVM hypervisor to provide a PTP device file that follows the system time of the host.
+Hyper-V also offers time synchronization through a PTP device file, managed by its _Linux Integration Services_.
+
+Paravirtualized time synchronization has the benefit of generally offering the highest precision.
+Like RTC clock emulation, it also works independently of the guest system's network connectivity.
+The main drawback is the dependency on a specific combination of hypervisor and guest operating system, because it requires hypervisor-specific drivers to run in the guest.
+
+## Decision
+
+From looking at the available options, it becomes apparent that there is no single optimal solution for time synchronization.
+The most precise option is the least portable, but the most widely supported option also requires the most provider-specific configuration.
+
+The aim of SCS is to improve interoperability between Openstack clouds, so the preferable standard solution is one that works across a wide range of setups.
+This perspective clearly favours NTP as a standardized method of time synchronization, because of it's wide support in cloud images and it's independence from other components, such as the hypervisor.
+
+However it is still useful for a CSP to offer less portable, but more precise methods of time synchronization, especially if they are integrated into the default cloud images offered by the CSP.
+
+So, we should standardize a portable NTP setup that users can assume when developing images to work well in any SCS cloud.
+We should not try to prevent CSPs from also supporting paravirtualized or other methods of time synchronization that may offer significant benefits over NTP.
+
+## Consequences
+
+A standardized NTP setup will allow SCS users and third party image providers to develop images that support local time synchronization across SCS clouds.
+
+As discussed in the NTP section above, there are a number of limitations in OpenStack and related projects that a standardized NTP setup has to work with:
+
+* OpenStack currently has no support for providing NTP servers to VMs under a fixed link-local address, like AWS and GCP are doing.
+  Such a feature could probably be implemented by re-using the metadata service IP (`169.254.169.254`) and the mechanisms to inject it into subnets.
+  If this feature becomes available in the future, it will be an attractive target for standardization, but until then it seems more sensible to focus on servers available through provider networks.
```

Comment:
It was afaik encouraged during the discussion to actually work on this upstream.

This will not be implemented upstream by just waiting for it.

At least we should mention this in the decision record imho.

Reply:
Sorry, I forgot to reply. I have since updated the DR accordingly.